### PR TITLE
API to retrieve top level messages

### PR DESCRIPTION
**Pull request by romanofski**

### CODE REVIEW

**Comment by romanofski** [comment]
After studying the notmuch API myself I'm now realising what you meant in purebred-mua/purebred#35 that notmuch has threading built in. Which makes this merge request probably not very useful.

Thinking about it we should make use of that, the `NotmuchMail` becomes a `NotmuchItem` and exposes a similar if not the same API for threads and mails (perhaps a different constructor?). A rendering class then can render different views of the item.

File: src/Notmuch.hs [hunk 1]
```diff
@@ -72,6 +72,7 @@ module Notmuch
   , queryCountThreads
 
   , Thread
+  , thread_top_messages
```
Comment (frasertweedale) [inline src/Notmuch.hs]: The main API should use camelCase. I think it should just be called `topLevelMessages`.

File: src/Notmuch/Binding.chs [hunk 2]
```diff
@@ -340,7 +340,13 @@ thread_get_thread_id ptr =
...
 
 -- notmuch_thread_get_total_messages
--- notmuch_thread_get_toplevel_messages -> Messages
+
+thread_get_toplevel_messages :: MonadIO m => Thread a -> m [Message 0 a]
+thread_get_toplevel_messages ptr = liftIO $ withThread ptr $ \ptr' ->
+  {#call thread_get_toplevel_messages #} ptr'
+    >>= detachPtr
+    >>= newForeignPtr messages_destroy
+    >>= messagesToList . Messages
```
Comment (frasertweedale) [inline src/Notmuch/Binding.chs:349]: LGTM

**Review by frasertweedale — CHANGES_REQUESTED** [review]
I assume you have tested the change?  If so, one nit (the name of the exposed function), then ACK.

File: src/Notmuch.hs (hunk 1 above)
Comment (romanofski) [reply src/Notmuch.hs]: Will change!

File: src/Notmuch/Binding.chs (hunk 2 above)
Comment (romanofski) [reply src/Notmuch/Binding.chs:349]: *phew*

File: src/Notmuch.hs [hunk 3]
```diff
@@ -142,10 +143,18 @@ instance HasMessages (Message n) where
   messages = message_get_replies
   -- replies!
 
-
 class HasThreads a where
```
Comment (frasertweedale) [inline src/Notmuch.hs:146]: i'll probably remove this class in a future commit.  but fine for now :)

**Review by frasertweedale — APPROVED** [review]

**Pull request merged**